modules/nlu/src/views/index.jsx

```diff
@@ -1,55 +1,64 @@
 import React from 'react'
-import classnames from 'classnames'
-import { Collapse, Button, Checkbox } from 'react-bootstrap'
+import {
+  Collapse,
+  Button,
+  Glyphicon,
+  ListGroup,
+  ListGroupItem,
+  FormGroup,
+  InputGroup,
+  FormControl
+} from 'react-bootstrap'
 import _ from 'lodash'
 
-import IntentEditor from './intents'
-import SyncConfirmModal from './sync'
+import EntitiesComponent from './entities'
 
+import IntentEditor from './intents'
 import style from './style.scss'
 
-const isHiddenIntent = ({ name }) => name.startsWith('__')
+// Exports SubViews
+export const entities = props => new EntitiesComponent(props)
 
 export default class Module extends React.Component {
   state = {
     showNavIntents: true,
-    showHiddenIntents: false,
     intents: [],
     currentIntent: null,
     filterValue: '',
-    syncNeeded: false,
-    showSyncConfirm: false
+    isSyncing: false
   }
 
   componentDidMount() {
     this.fetchIntents()
-
-    this.syncInterval = setInterval(this.checkSync, 10000)
     this.checkSync()
   }
 
-  componentWillUnmount() {
-    clearInterval(this.syncInterval)
+  sync = () => {
+    return this.props.bp.axios
+      .get('/mod/nlu/sync')
+      .catch(err => this.setState({ syncFailedError: err.response.data }))
+      .then(() => {
+        // Set animation timeout
+        setTimeout(() => this.setState({ isSyncing: false }), 2000)
+      })
   }
 
-  checkSync = () => {
-    return this.props.bp.axios.get('/mod/nlu/sync/check').then(res => {
-      if (this.state.syncNeeded !== res.data) {
-        this.setState({ syncNeeded: res.data })
+  checkSync = _.throttle(() => {
+    this.props.bp.axios.get('/mod/nlu/sync/check').then(res => {
+      if (res.data) {
+        this.setState({ isSyncing: true }, this.sync)
```

Comment:
why does `checkSync` syncs?  Is it a "sync if required" or is sync misnamed (it's a GET request so I'm confused if it actually syncs or not, if yes it should be a POST) ?

Reply:
`checkSync` will sync if the `/check` endpoint returns `true`. To actually sync we use another endpoint `sync`